### PR TITLE
[DRAFT] src: refactoring current implementation

### PR DESCRIPTION
**Pull request by minminlittleshrimp**
+ Remove dead code
+ Add NULL-variable and failed-process detection
+ Assign delimiters to const char type
+ Handle multithread cancellation

### CODE REVIEW

File: src/offlinelogstorage/dlt_offline_logstorage_behavior.c [hunk 1]
```diff
@@ -622,7 +622,10 @@ int dlt_logstorage_open_log_file(DltLogStorageFilterConfig *config,
                          "%s: Remove '%s' (num_log_files: %u, config->num_files:%u)\n",
                          __func__, absolute_file_path, num_log_files,
                          config->num_files);
-                remove(absolute_file_path);
+
+                if (remove(absolute_file_path) != 0)
```
Comment (lvklevankhanh) [inline src/offlinelogstorage/dlt_offline_logstorage_behavior.c:626]: Please apply this fix to check return at line #584

**Comment by lvklevankhanh** [comment]
Please check the line https://github.com/COVESA/dlt-daemon/blob/master/src/shared/dlt_multiple_files.c#L103
Should it be "newest" instead?

**Comment by lvklevankhanh** [comment]
Should we check file null before using at line?
https://github.com/COVESA/dlt-daemon/blob/master/src/offlinelogstorage/dlt_offline_logstorage_behavior.c#L430
For reference: 
https://github.com/COVESA/dlt-daemon/blob/22d394f4a633c7f08e13fd926339b879cb6f64da/src/kpi/dlt-kpi-common.c#L61

**Comment by lvklevankhanh** [comment]
Should we fix the coverity "Calling "malloc" which returns uninitialized memory" at lines?
https://github.com/COVESA/dlt-daemon/blob/master/src/daemon/dlt_daemon_common.c#L934
https://github.com/COVESA/dlt-daemon/blob/master/src/daemon/dlt_daemon_common.c#L946

**Comment by minminlittleshrimp** [comment]
Set to draft since lower prior

**Comment by minminlittleshrimp** [comment]
Close PR and replace by rebased PR: https://github.com/COVESA/dlt-daemon/pull/566

**Pull request closed without merging**